client/reader/recommend-button/index.tsx

```diff
@@ -1,52 +1,44 @@
 import { Button } from '@wordpress/components';
 import clsx from 'clsx';
 import { useTranslate } from 'i18n-calypso';
-import { useEffect } from 'react';
-import { useRecommendedSite } from 'calypso/landing/subscriptions/hooks/use-recommended-site';
-import { useDispatch, useSelector } from 'calypso/state';
+import { useCallback } from 'react';
+import { useRecommendedListMutation } from 'calypso/data/reader/recommendations/use-recommend-list-mutation';
+import { useSelector } from 'calypso/state';
 import { getCurrentUserName } from 'calypso/state/current-user/selectors';
-import { requestRecommendedBlogsListItems } from 'calypso/state/reader/lists/actions';
-import {
-	hasRequestedUserRecommendedBlogs,
-	isRequestingUserRecommendedBlogs,
-} from 'calypso/state/reader/lists/selectors';
 import ReaderFollowFeedIcon from '../components/icons/follow-feed-icon';
 import ReaderFollowingFeedIcon from '../components/icons/following-feed-icon';
 import type { Feed } from 'calypso/state/data-layer/wpcom/read/feed/types';
 import './style.scss';
 
 interface Props {
 	feedId: Feed[ 'feed_ID' ];
+	isRecommended: boolean;
 }
 
-export const RecommendButton = ( { feedId }: Props ) => {
-	const { isRecommended, toggleRecommended } = useRecommendedSite( feedId );
+export const RecommendButton = ( { feedId, isRecommended = false }: Props ) => {
 	const owner = useSelector( getCurrentUserName );
 	const translate = useTranslate();
-	const dispatch = useDispatch();
+	const { add, remove } = useRecommendedListMutation( owner );
```

Comment:
similarly, I  wonder if the toggleRecommended from https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/subscriptions/hooks/use-recommended-site.ts#L104 will work here.